core/src/com/unciv/logic/map/UnitPromotions.kt

```diff
@@ -15,8 +15,8 @@ class UnitPromotions{
     fun xpForNextPromotion() = (numberOfPromotions+1)*10
     fun canBePromoted() = XP >= xpForNextPromotion()
 
-    fun addPromotion(promotionName:String){
-        XP -= xpForNextPromotion()
+    fun addPromotion(promotionName:String, isFree:Boolean = false){
+        if (!isFree) XP -= xpForNextPromotion()
         promotions.add(promotionName)
         numberOfPromotions++
```

Comment:
This should also not happen if it's free, so it won't affect cost of future promotions